From 4c50859282ce4687a647a66e07043572ebf60f5d Mon Sep 17 00:00:00 2001
From: haiqiang yao <haiqiangx.yao@intel.com>
Date: Mon, 7 Dec 2015 12:19:09 +0800
Subject: [PATCH 07/13] atomisp2:only enable
 V4L2_CID_ATOMISP_SELECT_ISP_VERSION for BTNS

the patch will add to support conditional compilation for
ioctl V4L2_CID_ATOMISP_SELECT_ISP_VERSION.Current,the
V4L2_CID_ATOMISP_SELECT_ISP_VERSION will be used in BTNS
platform and no use on BYT/COHO platform.

Change-Id: Icecd8a7c6c71cf89a53677cf00d026351e07472e
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-13009
Original-Tracked-On: #H1504113061
Signed-off-by: haiqiang yao <haiqiangx.yao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/88483
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Build: Dessert M BXT RVP (SC)
Build: Dessert M CHT FFD (SC)
Build: Dessert M COHO (SC)
Validation-Android: Wei Tao <wei.tao@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Validation-Android: Dessert M COHO (SC)
Signed-off-by: Yang Fan F <fan.f.yang@intel.com>
Reviewed-on: https://android.intel.com:443/450388
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c | 2 ++
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c       | 5 ++++-
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.h       | 3 ++-
 3 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
index 9e4de4e..ce929f3 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
@@ -721,6 +721,7 @@ static void __apply_additional_pipe_config(
 
 		/* the isp default to use ISP2.2 and the camera hal will
 		 * control whether use isp2.7 */
+#ifdef V4L2_CID_ATOMISP_SELECT_ISP_VERSION
 		if (asd->select_isp_version->val ==
 			ATOMISP_CSS_ISP_PIPE_VERSION_2_7)
 			stream_env->pipe_configs[pipe_id].isp_pipe_version =
@@ -728,6 +729,7 @@ static void __apply_additional_pipe_config(
 		else
 			stream_env->pipe_configs[pipe_id].isp_pipe_version =
 				SH_CSS_ISP_PIPE_VERSION_2_2;
+#endif
 		break;
 	case IA_CSS_PIPE_ID_VIDEO:
 		/* enable reduced pipe to have binary
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c
index 63fce53..14dee6a 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c
@@ -1040,6 +1040,7 @@ static const struct v4l2_ctrl_config ctrl_depth_mode = {
  * isp will default to use ISP2.2.
  * Note: Make sure set this configuration before creating stream.
  */
+#ifdef V4L2_CID_ATOMISP_SELECT_ISP_VERSION
 static const struct v4l2_ctrl_config ctrl_select_isp_version = {
 	.ops = &ctrl_ops,
 	.id = V4L2_CID_ATOMISP_SELECT_ISP_VERSION,
@@ -1050,6 +1051,7 @@ static const struct v4l2_ctrl_config ctrl_select_isp_version = {
 	.step = 1,
 	.def = 0,
 };
+#endif
 
 #ifdef CONFIG_ION
 /*
@@ -1242,11 +1244,12 @@ static int isp_subdev_init_entities(struct atomisp_sub_device *asd)
 			v4l2_ctrl_new_custom(&asd->ctrl_handler,
 					     &ctrl_disable_dz,
 					     NULL);
+#ifdef V4L2_CID_ATOMISP_SELECT_ISP_VERSION
 	asd->select_isp_version =
 			v4l2_ctrl_new_custom(&asd->ctrl_handler,
 					     &ctrl_select_isp_version,
 					     NULL);
-
+#endif
 #ifdef CONFIG_ION
 	asd->ion_dev_fd =
 			v4l2_ctrl_new_custom(&asd->ctrl_handler,
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.h
index 80e533b..f388d3f 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.h
@@ -318,8 +318,9 @@ struct atomisp_sub_device {
 	struct v4l2_ctrl *enable_raw_buffer_lock;
 	struct v4l2_ctrl *ion_dev_fd;
 	struct v4l2_ctrl *disable_dz;
+#ifdef V4L2_CID_ATOMISP_SELECT_ISP_VERSION
 	struct v4l2_ctrl *select_isp_version;
-
+#endif
 	struct {
 		struct list_head fw;
 		struct list_head memory_maps;
-- 
1.9.1

